src/main/scala/sample/reactivekafka/KafkaReaderCoordinator.scala

```diff
@@ -41,19 +41,53 @@ class KafkaReaderCoordinator(mat: Materializer, topicName: String) extends Actor
       .kafkaOffsetsStorage()
       .commitInterval(1200 milliseconds)
     )
+
     log.debug("Starting the reader")
-    Source(consumerWithOffsetSink.publisher)
-      .map(processMessage)
+    val currencyRateSource = Source(consumerWithOffsetSink.publisher)
+    val currencyBroadcaster = Source(currencyRateSource.runWith(Sink.publisher(true)))
+
+    val alertsFlow = Flow[MessageAndMetadata[Array[Byte], CurrencyRateUpdated]]
+      .map(processAlert)
+      .filter(alert => alert != None)
+      .map(alert => alert.get)
+
+
+    // commit offset so we don't process the kafka message again
+    currencyBroadcaster
       .withAttributes(ActorAttributes.supervisionStrategy(processingDecider))
-      .to(consumerWithOffsetSink.offsetCommitSink).run()
```

Comment:
Ah right.
So the manual commit allow the message to be committed only if the condition is meet.
This would be very useful feature indeed.
May be I need to create another flow to demonstrate this.